cmd/config/subcommand/executionqueueattribute/file_config.go

```diff
@@ -0,0 +1,47 @@
+package executionqueueattribute
+
+import (
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+)
+
+// AttrFileConfig shadow Config for ExecutionQueueAttributes.
+// The shadow Config is not using ProjectDomainAttribute/Workflowattribute directly inorder to simplify the inputs.
+// As the same structure is being used for both ProjectDomainAttribute/Workflowattribute
+type AttrFileConfig struct {
+	Project  string `json:"project"`
+	Domain   string `json:"domain"`
+	Workflow string `json:"workflow,omitempty"`
+	*admin.ExecutionQueueAttributes
+}
+
+// Decorate decorator over ExecutionQueueAttributes.
+func (a AttrFileConfig) Decorate() *admin.MatchingAttributes {
+	return &admin.MatchingAttributes{
+		Target: &admin.MatchingAttributes_ExecutionQueueAttributes{
+			ExecutionQueueAttributes: a.ExecutionQueueAttributes,
+		},
+	}
+}
+
+// UnDecorate to uncover ExecutionQueueAttributes.
+func (a *AttrFileConfig) UnDecorate(matchingAttribute *admin.MatchingAttributes) {
+	if matchingAttribute == nil {
```

Comment:
when does this happen?

Reply:
This will happen in case where admin doesn't return an error of empty matching attribute and we try to unwrap the response to get the executionQueueAttribute which wont exist.
This is check is safeguard against it.

This is what is passed to undecorate 
`workflowAttr.GetAttributes().GetMatchingAttributes()`
Where workflowAttr is response from the admin 

Ideally it would be preferable to do null checks for both workflowAttr.GetAttributes() and then workflowAttr.GetAttributes().GetMatchingAttributes()

Since the idl can return nil for both
```

func (m *WorkflowAttributesGetResponse) GetAttributes() *WorkflowAttributes {
	if m != nil {
		return m.Attributes
	}
	return nil
}

...

func (m *WorkflowAttributes) GetMatchingAttributes() *MatchingAttributes {
	if m != nil {
		return m.MatchingAttributes
	}
	return nil
}```

Reply:
oh interesting, so there are cases where we don't return not found but just an empty response? that seems wrong

Reply:
We might be avoiding this in code path, but the idl supports returning nil hence added a guard .
I haven't seen it during my testing since if there are no attributes its always an error response and if thats the case then we don't call Undecorate.

If the resource is not found we do return an from the ORM layer
https://github.com/flyteorg/flyteadmin/blob/db32bb3810e7490e04ff693eadd26414aa294482/pkg/repositories/gormimpl/resource_repo.go#L114

And which is propogated back to client 
https://github.com/flyteorg/flyteadmin/blob/db32bb3810e7490e04ff693eadd26414aa294482/pkg/manager/impl/resources/resource_manager.go#L220

Hence i don't see how we can end up in this state no error and empty response.

Reply:
great, just wanted to confirm you hadn't actually seen this behavior! protecting against it since it's technically allowed by the response structure sounds fine